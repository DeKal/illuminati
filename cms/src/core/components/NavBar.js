import React from 'react'
import PropTypes from 'prop-types'
import { styled } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import IconButton from '@material-ui/core/IconButton'
import MenuIcon from '@material-ui/icons/Menu'
import Badge from '@material-ui/core/Badge'
import AccountCircle from '@material-ui/icons/AccountCircle'
import NotificationsIcon from '@material-ui/icons/Notifications'

const NavBar = ({ classes, history, open, openDrawer, closeDrawer }) => {
  return (
    <Container>
      <AppBar position="static" className={classes.navbar}>
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
            data-test-id="navbar-menu"
            onClick={() => (open ? closeDrawer() : openDrawer())}
          >
            <MenuIcon />
          </IconButton>
          <Typography
            variant="h1"
            data-test-id="navbar-title"
            className={classes.title}
            onClick={() => {
              history.push('/')
            }}
          >
            Dashboard
          </Typography>
          <div>
            <IconButton
              aria-label="show 17 new notifications"
              color="inherit"
              className={classes.icon}
            >
              <Badge badgeContent={17} color="secondary">
                <NotificationsIcon />
              </Badge>
            </IconButton>
            <IconButton
              edge="end"
              aria-label="account of current user"
              aria-haspopup="true"
              color="inherit"
              data-test-id="navbar-user-info"
              className={classes.icon}
              onClick={() => {
                history.push('/user')
              }}
            >
              <AccountCircle />
            </IconButton>
          </div>
        </Toolbar>
      </AppBar>
    </Container>
  )
}

export default NavBar

NavBar.propTypes = {
  classes: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  open: PropTypes.bool,
  openDrawer: PropTypes.func,
  closeDrawer: PropTypes.func
}

const Container = styled('div')({
  flexGrow: 1
})
