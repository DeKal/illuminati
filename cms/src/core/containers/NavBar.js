import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { withRouter } from 'react-router'
import { connect } from 'react-redux'
import NavBar from 'core/components/NavBar'
import { openDrawer, closeDrawer } from 'core/state/drawer/actions'

const useStyles = makeStyles(theme => ({
  menuButton: {
    marginRight: theme.spaces.gutter
  },
  title: {
    flexGrow: 1,
    float: 'left',
    fontWeight: 'bold',
    letterSpacing: '1px',
    textTransform: 'uppercase',
    cursor: 'pointer',
    fontSize: theme.fontSizes.XL,
    fontFamily: theme.fonts.montSerrat,
    color: theme.colors.navTitle
  },
  navbar: {
    height: `${theme.navbarHeight}px`,
    backgroundColor: theme.colors.primary
  },
  icon: {
    padding: theme.spaces.gutter
  }
}))

const NavBarContainer = props => <NavBar classes={useStyles()} {...props} />
const NavBarWithRouter = withRouter(NavBarContainer)

const mapStateToProps = state => ({
  open: state.drawer.open
})

const mapDispatchToProps = {
  openDrawer,
  closeDrawer
}

export default connect(mapStateToProps, mapDispatchToProps)(NavBarWithRouter)
