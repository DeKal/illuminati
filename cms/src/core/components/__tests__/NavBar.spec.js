import * as React from 'react'
import { shallow } from 'enzyme'
import NavBar from '../NavBar'

const classes = {
  navbar: '',
  menuButton: '',
  icon: ''
}

describe('NavBar', () => {
  it('should render NavBar when it opens', function() {
    const wrap = shallow(<NavBar classes={classes} open />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render NavBar when it closes', function() {
    const wrap = shallow(<NavBar classes={classes} open={false} />)
    expect(wrap).toMatchSnapshot()
  })

  it('click on menu should call openDrawer when it closes', function() {
    const openDrawer = jest.fn()
    const wrap = shallow(
      <NavBar classes={classes} open={false} openDrawer={openDrawer} />
    )
    wrap.find('[data-test-id="navbar-menu"]').simulate('click')
    expect(openDrawer.mock.calls.length).toBe(1)
  })

  it('click on menu should call closeDrawer when it opens', function() {
    const closeDrawer = jest.fn()
    const wrap = shallow(
      <NavBar classes={classes} open closeDrawer={closeDrawer} />
    )
    wrap.find('[data-test-id="navbar-menu"]').simulate('click')
    expect(closeDrawer.mock.calls.length).toBe(1)
  })

  it('click on navbar header should link to main page', function() {
    const history = {
      push: jest.fn()
    }
    const wrap = shallow(<NavBar classes={classes} open history={history} />)
    wrap.find('[data-test-id="navbar-title"]').simulate('click')
    expect(history.push.mock.calls.length).toBe(1)
    expect(history.push.mock.calls[0][0]).toBe('/')
  })

  it('click on navbar user icon should link to user page', function() {
    const history = {
      push: jest.fn()
    }
    const wrap = shallow(<NavBar classes={classes} open history={history} />)
    wrap.find('[data-test-id="navbar-user-info"]').simulate('click')
    expect(history.push.mock.calls.length).toBe(1)
    expect(history.push.mock.calls[0][0]).toBe('/user')
  })
})
