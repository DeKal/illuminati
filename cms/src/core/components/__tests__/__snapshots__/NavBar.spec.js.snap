// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render NavBar when it closes 1`] = `
<Styled(div)>
  <WithStyles(ForwardRef(AppBar))
    className=""
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        className=""
        color="inherit"
        data-test-id="navbar-menu"
        edge="start"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        data-test-id="navbar-title"
        onClick={[Function]}
        variant="h1"
      >
        Dashboard
      </WithStyles(ForwardRef(Typography))>
      <div>
        <WithStyles(ForwardRef(IconButton))
          aria-label="show 17 new notifications"
          className=""
          color="inherit"
        >
          <WithStyles(ForwardRef(Badge))
            badgeContent={17}
            color="secondary"
          >
            <NotificationsIcon />
          </WithStyles(ForwardRef(Badge))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          aria-haspopup="true"
          aria-label="account of current user"
          className=""
          color="inherit"
          data-test-id="navbar-user-info"
          edge="end"
          onClick={[Function]}
        >
          <AccountCircleIcon />
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</Styled(div)>
`;

exports[`NavBar should render NavBar when it opens 1`] = `
<Styled(div)>
  <WithStyles(ForwardRef(AppBar))
    className=""
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        className=""
        color="inherit"
        data-test-id="navbar-menu"
        edge="start"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        data-test-id="navbar-title"
        onClick={[Function]}
        variant="h1"
      >
        Dashboard
      </WithStyles(ForwardRef(Typography))>
      <div>
        <WithStyles(ForwardRef(IconButton))
          aria-label="show 17 new notifications"
          className=""
          color="inherit"
        >
          <WithStyles(ForwardRef(Badge))
            badgeContent={17}
            color="secondary"
          >
            <NotificationsIcon />
          </WithStyles(ForwardRef(Badge))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          aria-haspopup="true"
          aria-label="account of current user"
          className=""
          color="inherit"
          data-test-id="navbar-user-info"
          edge="end"
          onClick={[Function]}
        >
          <AccountCircleIcon />
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</Styled(div)>
`;
